src/docs/development/platform-integration/add-to-app/performance.md

```diff
@@ -0,0 +1,174 @@
+that the `FlutterEngine` is not yet attached to any UI components such as a
+`FlutterViewController` on iOS or a `FlutterActivity` on Android.
+
+### Attaching UI to the Flutter Engine
+
+A standard full-Flutter app moves to reach this state as soon as the app is
+launched.
+
+In an add-to-app scenario, this happens when you attach a `FlutterEngine`
+to a UI component such as by calling [`startActivity()`](https://developer.android.com/reference/android/content/Context.html#startActivity(android.content.Intent))
+with an [`Intent`](https://developer.android.com/reference/android/content/Intent.html)
+built via [`FlutterActivity.withCachedEngine()`]({{site.api}}/javadoc/io/flutter/embedding/android/FlutterActivity.html#withCachedEngine-java.lang.String-)
+on **Android**. Or by presenting a [`FlutterViewController`]({{site.api}}/objcdoc/Classes/FlutterViewController.html)
+initialized via [`initWithEngine: nibName: bundle:`]({{site.api}}/objcdoc/Classes/FlutterViewController.html#/c:objc(cs)FlutterViewController(im)initWithEngine:nibName:bundle:)
+on **iOS**.
+
+This would also the case if a Flutter UI component was launched without
+pre-warming a `FlutterEngine` such as via [`FlutterActivity.createDefaultIntent()`]({{site.api}}/javadoc/io/flutter/embedding/android/FlutterActivity.html#createDefaultIntent-android.content.Context-)
+on **Android** or via [`FlutterViewController initWithProject: nibName: bundle:`]({{site.api}}/objcdoc/Classes/FlutterViewController.html#/c:objc(cs)FlutterViewController(im)initWithProject:nibName:bundle:)
+on **iOS**. An implicit `FlutterEngine` is created in these cases. ;
+
+Behind the scene, both platform's UI components serves to provide the
+`FlutterEngine` with a rendering surface such as a [Surface](https://developer.android.com/reference/android/view/Surface)
+on **Android** or a [CAEAGLLayer](https://developer.apple.com/documentation/quartzcore/caeagllayer)
+or [CAMetalLayer](https://developer.apple.com/documentation/quartzcore/cametallayer)
+on **iOS**.
+
+At this point, the [Layer]({{site.api}}/flutter/rendering/Layer-class.html)
+tree generated by your Flutter program per frame will be converted into
+OpenGL (or Vulkan or Metal) GPU instructions.
+
+## Memory and latency
+
+Showing a Flutter UI has non-trivial latency cost. This cost can be alleviated
+by starting the Flutter engine ahead of time.
+
+The most relevant choice for users in add-to-app scenarios is to decide
+when to pre-load a `FlutterEngine` (that is, to load the Flutter library,
+start the Dart VM and run entrypoint in an Isolate) and the memory and latency
+cost of that pre-warm. And how it affects the memory and latency cost of
+rendering a first Flutter frame when UI is subsequently attached to that
+`FlutterEngine`.
+
+As of September of 2019, testing on a low-end 2015 class device in release-AOT
+mode, pre-warming the `FlutterEngine` costs:
+
+- 42MB and 1530ms to prewarm on **Android**. 330ms of it is a blocking call on
```

Comment:
330ms of it is a blocking => 330ms is due to a blocking

Reply:
Rather than trying to explain the previous number's composition, I'm just expanding on another subcomponent which is of interest in its own right